src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6706,17 +6706,47 @@ bool canBeNullableType()
                                 continue;
                         }
                         goto done; // token not consumed
+                    case SyntaxKind.OpenBracketToken when (nameOptions & NameOptions.PossiblePattern) != 0:
```

Comment:
We could possibly merge this with the case below, but it would complicate the normal path

Reply:
I think it's okay to keep separate if that helps for clarity, even if a bit of redundancy.

Reply:
Looking at this more, factoring the two blocks would probably still be understandable with `!inPattern || isOmittedSize()` as you suggested. Let's try that.